edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/entity/TransportCall.java

```diff
@@ -0,0 +1,47 @@
+package org.dcsa.edocumentation.domain.persistence.entity;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import lombok.*;
+import org.dcsa.skernel.domain.persistence.entity.base.BaseTransportCall;
+import org.hibernate.annotations.Formula;
+
+import javax.persistence.*;
+import java.time.OffsetDateTime;
+
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Setter(AccessLevel.PRIVATE)
+@Entity
+public class TransportCall extends BaseTransportCall {
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "vessel_id")
+  private Vessel vessel;
+
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "import_voyage_id")
+  private Voyage importVoyage;
+
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "export_voyage_id")
+  private Voyage exportVoyage;
+
+  @Formula("(select transport_event.event_date_time from transport_event where transport_event.transport_call_id = id AND" +
+    "  transport_event.event_classifier_code = 'PLN' AND   transport_event.transport_event_type_code = 'ARRI' )")
+  @JsonFormat(shape = JsonFormat.Shape.STRING)
+  private OffsetDateTime eventDateTimeArrival;
+
+  @Formula("(select transport_event.event_date_time from transport_event where transport_event.transport_call_id = id AND" +
+    "  transport_event.event_classifier_code = 'PLN' AND   transport_event.transport_event_type_code = 'DEPA' )")
+  @JsonFormat(shape = JsonFormat.Shape.STRING)
+  private OffsetDateTime eventDateTimeDeparture;
```

Comment:
This needs to fetch the "latest transport event" (of `PLN ARR/DEPA`) and get the `event_date_time` from that

Reply:
I have resolved this by utilizing Postgres defined `max()` function, which returns the latest timestamp

Reply:
Unfortunately, I do not think that is a correct solution because SQL is a bit counter intuitive here.

When the query does is take the latest arrival (or departure) time ever announced. What we want is to show the latest announced arrival (or departure) time.  These are not the same.

Consider this flow:

 1. Vessel is planned to arrive in Hamburg on in 7 days.
 2. Vessel is delayed and will now arrive in Hamburg in 13 days.
 3. Vessel omitted a port call (that was causing the delay) and the vessel will now arrive in 6 days instead.


The query as it is written will return `13 days` (option 2) because that is the "highest" arrival time, but we should return `6 days` (option 3) which is the latest announced arrival time.

Reply:
Yes, I see what you mean. I will have another look at it, thanks

Reply:
I tried again with the `@Formula `  to avoid adding entities. If not this, then I think creating a view might be the solution.

Reply:
Looks like they might work now. I have not tested them, but I assume you have. :)

Reply:
I indeed did